plotly/tools.py

```diff
@@ -1443,6 +1449,398 @@ class FigureFactory(object):
+        given parametrized surface. It returns an rgb color based on the
+        relative position of zval between vmin and vmax
+        """
+        if vmin >= vmax:
+            raise exceptions.PlotlyError("Incorrect relation between vmin "
+                                         "and vmax. The vmin value cannot be "
+                                         "bigger than or equal to the value "
+                                         "of vmax.")
+        # find distance t of zval from vmin to vmax where the distance
+        # is normalized to be between 0 and 1
+        t = (zval - vmin)/float((vmax - vmin))
+        t_color = FigureFactory._find_intermediate_color(colormap[0],
+                                                         colormap[1],
+                                                         t)
+        t_color = (t_color[0]*255.0, t_color[1]*255.0, t_color[2]*255.0)
+        labelled_color = 'rgb{}'.format(t_color)
+
+        return labelled_color
+
+    @staticmethod
+    def _tri_indices(simplices):
+        """
+        Returns a triplet of 3 lists such that each list contains the kth
+        index of each triplet of the simplicies, where k is 0, 1 and 2
+        """
+        return ([triplet[c] for triplet in simplices] for c in range(3))
+
+    @staticmethod
+    def _trisurf(x, y, z, simplices, colormap=None,
+                 plot_edges=None, Xe=None, Ye=None, Ze=None):
+        """
+        Refer to FigureFactory.create_trisurf() for docstring
+        """
+        # numpy import check
+        if _numpy_imported is False:
+            raise ImportError("FigureFactory._trisurf() requires "
+                              "numpy imported.")
+        import numpy as np
+        from plotly.graph_objs import graph_objs
+        points3D = np.vstack((x, y, z)).T
+
+        # vertices of the surface triangles
+        tri_vertices = list(map(lambda index: points3D[index], simplices))
+        # mean values of z-coordinates of triangle vertices
+        zmean = [np.mean(tri[:, 2]) for tri in tri_vertices]
+        min_zmean = np.min(zmean)
+        max_zmean = np.max(zmean)
+        facecolor = ([FigureFactory._map_z2color(zz,  colormap, min_zmean,
+                      max_zmean) for zz in zmean])
+        I, J, K = FigureFactory._tri_indices(simplices)
+
+        triangles = graph_objs.Mesh3d(x=x, y=y, z=z, facecolor=facecolor,
+                                      i=I, j=J, k=K, name='')
+
+        if plot_edges is None:  # the triangle sides are not plotted
+            return graph_objs.Data([triangles])
+
+        # define the lists Xe, Ye, Ze, of x, y, resp z coordinates of
+        # edge end points for each triangle
+        # None separates data corresponding to two consecutive triangles
+        lists_coord = ([[[T[k % 3][c] for k in range(4)]+[None]
+                        for T in tri_vertices] for c in range(3)])
+        if Xe is None:
```

Comment:
🐄 try not to use `TitleCase` for anything but `class`es in Python. Can we rename this something like `x_edge`?